torch/csrc/jit/codegen/cuda/lower2device.cpp

```diff
@@ -434,17 +435,17 @@ std::vector<Expr*> GPULower::getLoweredExprs() {
         mutated_stmt->isExpr(),
         "Tried to generate a kernel but hit a non expression during lowering: ",
         mutated_stmt);
-    lowered_exprs.push_back(static_cast<Expr*>(mutated_stmt));
+    exprs.push_back(static_cast<Expr*>(mutated_stmt));
   }
 
-  return lowered_exprs;
+  return exprs;
 }
 
 std::ostream& GPULower::printKernel(
     std::ostream& os,
     const std::string& kernel_name) {
-  FusionGuard fg(fusion_);
-  getLoweredExprs();
+  FusionGuard fg(fusion_, nullptr);
+  lowered_exprs = getLoweredExprs(); // was this the intention?
```

Comment:
I think lowered_exprs was a remnant, we could probably just remove it unless I'm missing something

Reply:
it's just a nit - we can do it either way, but if we don't want to set it here I don't think it makes sense to return it from getLoweredExprs()

Reply:
I had lowered_exprs as a member because I was originally reusing it, however, I think it's better to recompile since we can now do that in case the fusion gets modified in between compilations.